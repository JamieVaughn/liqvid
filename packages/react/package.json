{
  "name": "@liqvid/react",
  "version": "0.0.1",
  "description": "React surface for Liqvid",
  "exports": {
    ".": "./dist/index.js"
  },
  "typesVersions": {
    "*": {
      "*": ["./dist/*"]
    }
  },
  "files": ["dist/*"],
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/liqvidjs/liqvid.git"
  },
  "author": "Yuri Sulyma <yuri@liqvidjs.org>",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/liqvidjs/liqvid/issues"
  },
  "homepage": "https://github.com/liqvidjs/liqvid#readme",
  "dependencies": {
    "@types/events": "^3.0.0",
    "@types/react-reconciler": "^0.26.4"
  },
  "devDependencies": {
    "events": "^3.3.0",
    "strict-event-emitter-types": "^2.0.0"
  },
  "peerDependencies": {
    "liqvid": "workspace:^",
    "react": ">=17.0.0",
    "react-dom": ">=17.0.0"
  },
  "sideEffects": false
}
