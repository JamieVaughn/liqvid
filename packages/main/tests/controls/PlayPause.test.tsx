import * as React from "react";
import {fireEvent, render} from "@testing-library/react";

import "../matchMedia.mock";
import "../DocumentTimeline.mock";

import {Playback, Player} from "../..";
import {act} from "react-dom/test-utils";

describe("Play/pause button", () => {
  let button: HTMLButtonElement;
  
  const playback = new Playback({duration: 60000});

  beforeEach(() => {
    render(<Player playback={playback}></Player>);
    act(() => {
      playback.pause();
      playback.seeking = false;
    });
    button = document.querySelector(".lv-controls-playpause > svg");
  });

  test("Clicking button toggles", () => {
    fireEvent.click(button);
    expect(playback.paused).toBe(false);
    fireEvent.click(button);
    expect(playback.paused).toBe(true);
  });

  test("Icon updates", () => {
    expect(button).toMatchSnapshot();
    act(() => {
      playback.play();
    });
    expect(button).toMatchSnapshot();
    act(() => {
      playback.seeking = true;
    });
    expect(button).toMatchSnapshot();
  });

  test("Keyboard shortcuts work", () => {
    fireEvent.keyDown(document.body, {key: "K", code: "KeyK"});
    expect(playback.paused).toBe(false);

    fireEvent.keyDown(document.body, {key: "K", code: "KeyK"});
    expect(playback.paused).toBe(true);

    fireEvent.keyDown(document.body, {key: " ", code: "Space"});
    expect(playback.paused).toBe(false);

    fireEvent.keyDown(document.body, {key: " ", code: "Space"});
    expect(playback.paused).toBe(true);
  });
});
