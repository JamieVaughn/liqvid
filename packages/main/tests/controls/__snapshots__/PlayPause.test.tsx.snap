// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Play/pause button Icon updates 1`] = `
<svg
  viewBox="0 0 36 36"
>
  <path
    d="M 12,26 18.5,22 18.5,14 12,10 z M 18.5,22 25,18 25,18 18.5,14 z"
    fill="white"
  />
</svg>
`;

exports[`Play/pause button Icon updates 2`] = `
<svg
  viewBox="0 0 36 36"
>
  <path
    d="M 12 26 h 4 v -16 h -4 z M 21 26 h 4 v -16 h -4 z"
    fill="white"
  />
</svg>
`;

exports[`Play/pause button Icon updates 3`] = `
<svg
  viewBox="0 0 36 36"
>
  <path
    d="M 12,26 18.5,22 18.5,14 12,10 z M 18.5,22 25,18 25,18 18.5,14 z"
    fill="white"
  />
</svg>
`;
